src/main/java/br/com/poupeAi/service/DespesaService.java

```diff
@@ -0,0 +1,94 @@
+package br.com.poupeAi.service;
+
+import br.com.poupeAi.exception.*;
+import br.com.poupeAi.model.*;
+import br.com.poupeAi.repository.PlanejamentoMensalRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+public class DespesaService {
+    private final PlanejamentoMensalRepository planejamentoRespository;
+    private final PlanejamentoMensalService planejamentoService;
+
+    @Autowired
+    public DespesaService(PlanejamentoMensalRepository planejamentoRepository,
+                          PlanejamentoMensalService planejamentoService) {
+        this.planejamentoRespository = planejamentoRepository;
+        this.planejamentoService = planejamentoService;
+    }
+
+    public void validarDespesa(PlanejamentoMensal planejamento, Envelope envelope, Despesa despesa)
+            throws NegocioException {
+        double saldoEnvelope = envelope.getOrcamento();
+        saldoEnvelope -= envelope.getDespesas()
+                .stream()
+                .filter(despesa1 -> !despesa1.getId().equals(despesa.getId()))
+                .mapToDouble(Despesa::getQuantia)
+                .sum();
+
+        if(saldoEnvelope < despesa.getQuantia())
+            throw new NegocioException("Saldo insuficiente no envelope. Saldo: R$ " + saldoEnvelope);
+
+        if(despesa.isEhParaOutroEnvelope() && Objects.isNull(despesa.getEnvelope()))
+            throw new NegocioException("O envelope para o qual está sendo emprestada a quantia deve ser indicado.");
+        else if(Objects.nonNull(despesa.getEnvelope())) {
+            Envelope outroEnvelope = planejamentoService.buscarEnvelopeNoPlanejamento(planejamento, despesa.getEnvelope().getId());
+            despesa.setEnvelope(outroEnvelope);
```

Comment:
essas validações são necessárias? Porque pelo que entendi no `DespesaInputDto` você definiu que o envelope não será nulo, né?

Reply:
Na verdade, o envelope pode ser nulo no `DespesaInputDto`, mas se ele não for, seu id também não poderá ser nulo.